### DISCUSSION TITLE
A new feature for Activation Event Hook

### DISCUSSION THREAD

**Discussion by yehuuu6** [SmartlyDressedGames/Unturned-3.x-Community#4351] [Ideas]
I've been creating mods since 2018 I think, and the most challenging aspects have involved dealing with multiplayer functionality. Some actions weren't triggering on the server due to missing features in mod hooks. Let's talk about the activation event hook. Currently, it executes its orders on either the server or the client, depending on the game object it's attached to. For instance, if I attach it to the Siren game object, it fires the events on the server; otherwise, it only triggers them for the client. This is making modders' lives difficult. Perhaps we could consider adding a new option to the activation event hook, such as "use on server," which ensures that the event is fired on the server when checked. Are there any plans for implementing something like this in the future?
Or you already did and I'm just unaware of it?

**Comment by GazziFX** [comment]
It doesn't work that way you just need to put it on both client and server prefabs

**Reply by yehuuu6** [reply]
By server prefabs do you mean clip? because i dont use it. Im trying to create systems that work with UI buttons, but those buttons do not fire events on server that is the issue here. Im trying to find a solution for that. You can use Sirens, Lock/Unlock functions for server events but there are only 3 or max 4 ways of doing it which is not enough "server prefabs" thats why maybe in the future we can add a new event hook. It's just my theory, i might be completely wrong.

**Reply by GazziFX** [reply]
Synchronization of custom objects not planned IIRC, to change state from client you can use InteractableObjectBinaryState with IOBS mod hook and set ToggleState() to UI button event

**Reply by GazziFX** [reply]
You can also attach `PluginButtonListener` to handle your buttons from plugins, and fire events that you actually needed to

**Comment by SDGNelson** [comment]
From the original post I at first though you wanted an option like "Authority Only" for Activation Event Hook, but judging from your other comment it sounds like what you're asking for is the ability to send custom events to the server? One way to do this currently is the chat messenger stuff, but those require the server owner to opt-in.
